src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,119 @@
+package duke.parser;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import duke.task.Task;
+import duke.task.ToDo;
+import duke.task.Event;
+import duke.task.Deadline;
+import duke.command.Command;
+import duke.command.ListCommand;
+import duke.command.MarkCommand;
+import duke.command.UnmarkCommand;
+import duke.command.ToDoCommand;
+import duke.command.DeadlineCommand;
+import duke.command.EventCommand;
+import duke.command.FindCommand;
+import duke.command.DeleteCommand;
+import duke.command.ByeCommand;
+import duke.command.InvalidCommand;
+import duke.ui.Ui;
+
+/**
+ * Handles parsing of user input and file input.
+ */
+public class Parser {
+    Ui ui;
+
+    /**
+     * Initialises the parser object with a Ui object.
+     * 
+     * @param ui The Ui object to handle user interaction.
+     */
+    public Parser(Ui ui) {
+        this.ui = ui;
+    }
+
+    /**
+     * Parses the task from the file and returns a Task object.
+     * 
+     * @param fileTask The task in the file.
+     * @return A Task object.
+     */
+    public Task parseFileTask(String fileTask) {
+        String[] taskComponents = fileTask.split(" \\| ");
+        String taskType = taskComponents[0];
+        boolean taskStatus = taskComponents[1].equals("1");
```

Comment:
Since it is a boolean type, how about considering change the name to something like isMarked?